packages/http/src/validator/__tests__/functional.spec.ts

```diff
@@ -40,6 +41,45 @@ describe('HttpValidator', () => {
           }),
         ).toMatchSnapshot();
       });
+
+      it('is case insensitive', async () => {
+        expect(
+          await validator.validateInput({
+            resource: {
+              method: 'GET',
+              path: '/hey',
+              responses: [
+                {
+                  code: '200',
+                },
+              ],
+              id: 'hey',
+              request: {
+                headers: [
+                  {
+                    name: 'API_KEY',
+                    style: HttpParamStyles.Simple,
+                    schema: {
+                      type: 'string',
+                    },
+                    required: true,
+                  },
+                ],
+              },
+            },
+            input: {
+              method: 'get',
+              url: {
+                path: '/hey',
+              },
+              headers: {
+                api_key: 'ha',
```

Comment:
If I'm not mistaken this test will only verify that the request headers are case insensitive, but not that the `input.headers` are. 

E.g. this does not confirm that `input.headers.Api_key` works fine.

In theory the actual code could looks something like this:
```ts
if (request.headers.name.toLowerCase() === Object.keys(input.headers)[0])
```

Reply:
This is a good point — the reason I didn't handle this case is that the nodejs HTTP server will *always* put the headers in lowercase, so this is a scenario that should never happen.

That said, given we will probably use Prism directly from the client, it might indeed make sense to make this normalisation in Prism itself. Let me check it out and I'll probably put another commit to fix this.

Good catch! 👍

Reply:
> the nodejs HTTP server will always put the headers in lowercase

Every server or just the current implementation? (rhetorical question I guess, since it must be implementation specific).

Yeah like you said I think it still makes sense to patch/test it here.

Reply:
Every server done in NodeJs will have lowercased headers.

https://nodejs.org/api/http.html#http_message_headers

> Header names are lower-cased.